### DISCUSSION TITLE
Need help to create template to collate annotations from multiple items based on colour

### DISCUSSION THREAD

**Discussion by sicahjes** [windingwind/zotero-better-notes#1237] [Q&A]
I'm hoping someone might be able to help me adjust this to what I need. Each time I try I break it. 

If possible, I'd like to adjust the 'slightly modified version of annotation collection by color' template, per [comment 406](https://github.com/windingwind/zotero-better-notes/discussions/406#discussioncomment-6221278) to achieve the below:

- Collates all annotations of selected items based on colour, 
- Each grouping should be in a section with a heading of the colour (eg Yellow, Purple etc.). 
- Each heading should be highlighted with the corresponding colour. 
- Ideally: there would be a section at the end with a heading called ‘Bibliography’ that provides the full citation for each source that features in the note, listed once and ordered alphabetically by surname. 

I'd also like to resolve whatever the issue is that results in a blank note when multiple items are selected. I'm thinking it is because there's no attachment or annotation in some of them, but I'm not 100% certain which contain notes (thus the need for the template) and I'm trying to compile them.

I've looked at the documentation and tried creating a template from scratch, but I'm conscious of a terrible looming deadline (that this template would make much easier) and I think my general stress is getting in the way of absorbing the information. 

I've really tried to do this without help, including different templates: some work as described, some return an invalid error, and some include random text at the top (IDs etc). I'd be grateful for anyone that could help me!

**Comment by fabianlatz** [comment]
@sicahjes I've had the same use case and came up with a solution myself. It should match all your requirements except the bibliography which would have produced a lot more workload for me and I don't find it useful due to the exhaustive generation features for bibliographies by Zotero itself. You can find it here: #1272 
I hope, this still comes in time as I just discovered your question yesterday on the lookout for a template for myself… wish you all the best for your research!
